version: '3.8'
services:
  fastapi:
    build:
      context: .
      dockerfile: ./docker/api/Dockerfile
    ports:
      - "8000:80"
    volumes:
      - ./api:/app/api
    env_file:
      - .env
    environment:
      - PYTHONUNBUFFERED=1
    depends_on:
      - db
    networks:
      - backend

  db:
    image: mysql:8.0
    platform: linux/amd64
    command: --default-authentication-plugin=mysql_native_password
    volumes:
      - ./docker/db/conf.d:/etc/mysql/conf.d
      - ./docker/db/initdb.d:/docker-entrypoint-initdb.d
      - db-data:/var/lib/mysql
    env_file:
      - .env
    ports:
      - "3306:3306"
    networks:
      - backend

  test_db:
    image: mysql:8.0
    platform: linux/amd64
    command: --default-authentication-plugin=mysql_native_password
    volumes:
      - ./docker/test_db/conf.d:/etc/mysql/conf.d
      - ./docker/test_db/initdb.d:/docker-entrypoint-initdb.d
      - test-db-data:/var/lib/mysql
    env_file:
      - .env
    ports:
      - "3307:3307"
    networks:
      - backend

volumes:
  db-data:
  test-db-data:

networks:
  backend:
    driver: bridge
